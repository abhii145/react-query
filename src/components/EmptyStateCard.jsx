import { FaShoppingBag } from "react-icons/fa"
import { Link } from "react-router"
import PropTypes from "prop-types"

const EmptyStateCard = ({
  icon: Icon = FaShoppingBag,
  heading,
  subheading,
  buttonLabel,
}) => {
  return (
    <div className="flex flex-col items-center justify-center bg-white rounded-md shadow-[0_2px_12px_-3px_rgba(6,81,237,0.3)] p-8 space-y-4 max-w-md mx-auto mt-20">
      <div className="p-4 bg-blue-50 text-blue-600 rounded-full">
        <Icon size={48} />
      </div>

      <div className="text-center">
        <h2 className="text-xl font-bold text-gray-800">{heading}</h2>
        <p className="text-sm text-gray-500">{subheading}</p>
      </div>

      <Link to="/">
        <button className="px-6 py-2 text-sm font-semibold text-white bg-gray-800 rounded-md hover:bg-gray-900">
          {buttonLabel}
        </button>
      </Link>
    </div>
  )
}
EmptyStateCard.propTypes = {
  icon: PropTypes.elementType,
  heading: PropTypes.string.isRequired,
  subheading: PropTypes.string,
  buttonLabel: PropTypes.string.isRequired,
}

export default EmptyStateCard
