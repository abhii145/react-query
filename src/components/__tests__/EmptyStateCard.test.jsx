import { render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import EmptyStateCard from "../EmptyStateCard"
import { it, expect } from "vitest"

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <EmptyStateCard
        heading="No items found"
        subheading="Try adding some items to your cart."
        buttonLabel="Go to Shop"
      />
    </MemoryRouter>
  )
  expect(asFragment()).toMatchSnapshot()
})
it("renders EmptyStateCard component", () => {
  render(
    <MemoryRouter>
      <EmptyStateCard
        heading="No items found"
        subheading="Try adding some items to your cart."
        buttonLabel="Go to Shop"
      />
    </MemoryRouter>
  )

  expect(screen.getByText("No items found")).toBeInTheDocument()
  expect(
    screen.getByText("Try adding some items to your cart.")
  ).toBeInTheDocument()
  expect(
    screen.getByRole("button", { name: /Go to Shop/i })
  ).toBeInTheDocument()
})
