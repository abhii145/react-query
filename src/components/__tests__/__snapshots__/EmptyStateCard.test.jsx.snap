// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center bg-white rounded-md shadow-[0_2px_12px_-3px_rgba(6,81,237,0.3)] p-8 space-y-4 max-w-md mx-auto mt-20"
  >
    <div
      class="p-4 bg-blue-50 text-blue-600 rounded-full"
    >
      <svg
        fill="currentColor"
        height="48"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M352 160v-32C352 57.42 294.579 0 224 0 153.42 0 96 57.42 96 128v32H0v272c0 44.183 35.817 80 80 80h288c44.183 0 80-35.817 80-80V160h-96zm-192-32c0-35.29 28.71-64 64-64s64 28.71 64 64v32H160v-32zm160 120c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zm-192 0c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24z"
        />
      </svg>
    </div>
    <div
      class="text-center"
    >
      <h2
        class="text-xl font-bold text-gray-800"
      >
        No items found
      </h2>
      <p
        class="text-sm text-gray-500"
      >
        Try adding some items to your cart.
      </p>
    </div>
    <a
      data-discover="true"
      href="/"
    >
      <button
        class="px-6 py-2 text-sm font-semibold text-white bg-gray-800 rounded-md hover:bg-gray-900"
      >
        Go to Shop
      </button>
    </a>
  </div>
</DocumentFragment>
`;
